src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -74,8 +74,8 @@ const ioTask = ({ name }) => _.nth(-2, name.split('.'))
 const ioVariable = ({ name }) => _.nth(-1, name.split('.'))
 const ioType = ({ inputType, outputType }) => (inputType || outputType).match(/(.*?)\??$/)[1] // unify, and strip off trailing '?'
 
-const WorkflowIOTable = ({ which, inputsOutputs, config, errors, onChange, onSetDefaults, suggestions }) => {
-  const data = inputsOutputs[which]
+const WorkflowIOTable = ({ which, inputsOutputs, config, errors, onChange, onSetDefaults, suggestions, includeOptionalInputs }) => {
+  const data = which === 'inputs' ? _.filter(includeOptionalInputs ? (() => true) : { optional: false }, inputsOutputs[which]) : inputsOutputs[which]
```

Comment:
nit: since both of these ternaries are effectively deciding whether to filter, the logic could be combined

Reply:
Agreed. I find the nested ternaries hard to read. I feel like there should also be a way to only express `inputsOutputs[which]` once and have it optionally filtered but I can't come up with the syntax at the moment.